File: src/test/java/no/nav/syfo/util/LdapUtil.java
package no.nav.syfo.util;

import no.nav.syfo.domain.AdRoller;
import no.nav.syfo.services.LdapService;

import static org.mockito.Mockito.reset;
import static org.mockito.Mockito.when;

public class LdapUtil {

    public static void mockRoller(LdapService ldapService, String veileder, boolean innvilget, AdRoller... roller) {
        reset(ldapService);

        for(AdRoller rolle : roller) {
            when(ldapService.harTilgang(veileder, rolle.rolle)).thenReturn(innvilget);
        }
    }

}

File: src/test/kotlin/no/nav/syfo/testhelper/LdapUtil.kt
package no.nav.syfo.testhelper

import no.nav.syfo.consumer.ldap.LdapService
import no.nav.syfo.domain.*
import org.mockito.Mockito

object LdapUtil {
    @JvmStatic
    fun mockRoller(
        ldapService: LdapService,
        veileder: String,
        innvilget: Boolean,
        vararg roller: AdRolle,
    ) {
        Mockito.reset(ldapService)
        for (rolle in roller) {
            Mockito.`when`(ldapService.harTilgang(veileder, rolle)).thenReturn(innvilget)
        }
    }
}
